core/shared/src/test/scala/com/softwaremill/sttp/testing/HttpTest.scala

```diff
@@ -249,37 +255,43 @@ trait HttpTest[R[_]]
   protected def withTemporaryFile[T](content: Option[Array[Byte]])(f: SttpFile => Future[T]): Future[T]
   private def withTemporaryNonExistentFile[T](f: SttpFile => Future[T]): Future[T] = withTemporaryFile(None)(f)
 
-  protected def sha256Hash(bytes: Array[Byte]): String
-  protected def sha256FileHash(file: SttpFile): String
+  protected def md5Hash(bytes: Array[Byte]): String
+  protected def md5FileHash(file: SttpFile): Future[String]
 
   "download file" - {
 
     "download a binary file using asFile" in {
       withTemporaryNonExistentFile { file =>
         val req = sttp.get(uri"$endpoint/download/binary").response(asSttpFile(file))
-        req.send().toFuture().map { resp =>
-          sha256FileHash(resp.unsafeBody) shouldBe binaryFileHash
+        req.send().toFuture().flatMap { resp =>
+          md5FileHash(resp.unsafeBody).map { _ shouldBe binaryFileHash }
         }
       }
     }
 
     "download a text file using asFile" in {
       withTemporaryNonExistentFile { file =>
         val req = sttp.get(uri"$endpoint/download/text").response(asSttpFile(file))
-        req.send().toFuture().map { resp =>
-          sha256FileHash(resp.unsafeBody) shouldBe textFileHash
+        req.send().toFuture().flatMap { resp =>
+          md5FileHash(resp.unsafeBody).map { _ shouldBe textFileHash }
         }
       }
     }
   }
 
+  protected def multipartStringDefaultFileName: Option[String] = None
+  private def defaultFileName = multipartStringDefaultFileName match {
+    case None => ""
+    case Some(name) => s" ($name)"
+  }
+
   "multipart" - {
     def mp = sttp.post(uri"$endpoint/multipart")
 
     "send a multipart message" in {
       val req = mp.multipartBody(multipart("p1", "v1"), multipart("p2", "v2"))
       req.send().toFuture().map { resp =>
-        resp.unsafeBody should be("p1=v1, p2=v2")
+        resp.unsafeBody should be(s"p1=v1$defaultFileName, p2=v2$defaultFileName")
```

Comment:
I don't think I understand why this is needed - maybe a comment which would explain?

Reply:
There is a comment in the JavaScript `HttpTest`, I'll move it into here.